Creating `r/virtual_machine` with multiple PCI passthrough devices fails

### Community Guidelines

- [X] I have read and agree to the [HashiCorp Community Guidelines ](https://www.hashicorp.com/community-guidelines).
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [ ] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform

1.1

### Terraform Provider

2.2.0

### VMware vSphere

7.0.3

### Description

if I create VM with more than 1 PCI device I receive the following error:

```console
error reconfiguring virtual machine: error reconfiguring virtual machine: A specified parameter was not correct: deviceChange[4].device.key
```

I enabled debug and Ii see the following request XML with the same key value for all devices.

### Affected Resources

`r/vsphere_virtual_machine`

### Terraform Configuration

_No response_

### Debug Output

```xml
<deviceChange xmlns:XMLSchema-instance="http://www.w3.org/2001/XMLSchema-instance" XMLSchema-instance:type="VirtualDeviceConfigSpec">
          <operation>add</operation>
          <device XMLSchema-instance:type="VirtualPCIPassthrough">
            <key>0</key>
            <backing XMLSchema-instance:type="VirtualPCIPassthroughDeviceBackingInfo">
              <deviceName/>
              <id>0000:5e:00.0</id>
              <deviceId/>
              <systemId>626059a0-e234-ce2c-a1b3-5c6f69787340</systemId>
              <vendorId>4318</vendorId>
            </backing>
          </device>
        </deviceChange>
        <deviceChange xmlns:XMLSchema-instance="http://www.w3.org/2001/XMLSchema-instance" XMLSchema-instance:type="VirtualDeviceConfigSpec">
          <operation>add</operation>
          <device XMLSchema-instance:type="VirtualPCIPassthrough">
            <key>0</key>
            <backing XMLSchema-instance:type="VirtualPCIPassthroughDeviceBackingInfo">
              <deviceName/>
              <id>0000:60:00.0</id>
              <deviceId/>
              <systemId>626059a0-e234-ce2c-a1b3-5c6f69787340</systemId>
              <vendorId>4318</vendorId>
            </backing>
          </device>
        </deviceChange>
        <deviceChange xmlns:XMLSchema-instance="http://www.w3.org/2001/XMLSchema-instance" XMLSchema-instance:type="VirtualDeviceConfigSpec">
          <operation>add</operation>
          <device XMLSchema-instance:type="VirtualPCIPassthrough">
            <key>0</key>
            <backing XMLSchema-instance:type="VirtualPCIPassthroughDeviceBackingInfo">
              <deviceName/>
              <id>0000:86:00.0</id>
              <deviceId/>
              <systemId>626059a0-e234-ce2c-a1b3-5c6f69787340</systemId>
              <vendorId>4318</vendorId>
            </backing>
          </device>
        </deviceChange>
```

### Panic Output

_No response_

### Expected Behavior

`r/virtual_machine` created with multiple PCI passthrough devices.

### Actual Behavior

Fails with error.

### Steps to Reproduce

_No response_

### Environment Details

_No response_

### Screenshots

_No response_

### References

_No response_

@tenthirtyam  / @wladikz,
I just hit the same issue using vSphere provider 2.2.0. Trying to assign 2 GPUs using PCI Passthrough seems not to work. 

````bash
2023-05-04T12:30:37.8139322Z [33;1mVERBOSE: 2023-05-04T12:30:44.399Z [INFO]  provider.terraform-provider-vsphere_v2.2.0_x5: 2023/05/04 12:30:44 [DEBUG] VM "/VxRail-Datacenter/vm/VMSK8G02" found for UUID "421c6936-12e4-218a-2657-53b29194ae39": timestamp=2023-05-04T12:30:44.399Z[0m
2023-05-04T12:30:37.8140560Z [33;1mVERBOSE: 2023-05-04T12:30:44.400Z [INFO]  provider.terraform-provider-vsphere_v2.2.0_x5: 2023/05/04 12:30:44 [DEBUG] Fetching properties for VM "/VxRail-Datacenter/vm/VMSK8G02": timestamp=2023-05-04T12:30:44.400Z[0m
2023-05-04T12:30:37.8141635Z [33;1mVERBOSE: 2023-05-04T12:30:44.404Z [INFO]  provider.terraform-provider-vsphere_v2.2.0_x5: 2023/05/04 12:30:44 [DEBUG] applyDeviceChange: Applying changes: (add: *types.VirtualPCIPassthrough at key 0): timestamp=2023-05-04T12:30:44.404Z[0m
2023-05-04T12:30:37.8142926Z [33;1mVERBOSE: 2023-05-04T12:30:44.404Z [INFO]  provider.terraform-provider-vsphere_v2.2.0_x5: 2023/05/04 12:30:44 [DEBUG] applyDeviceChange: Device list before changes: ide-200,ide-201,ps2-300,pci-100,sio-400,keyboard-600,pointing-700,video-500,vmci-12000,pvscsi-1000,disk-1000-0,disk-1000-1,ethernet-0,pcipassthrough-0: timestamp=2023-05-04T12:30:44.404Z[0m
2023-05-04T12:30:37.8144238Z [33;1mVERBOSE: 2023-05-04T12:30:44.404Z [INFO]  provider.terraform-provider-vsphere_v2.2.0_x5: 2023/05/04 12:30:44 [DEBUG] applyDeviceChange: Device list after changes: ide-200,ide-201,ps2-300,pci-100,sio-400,keyboard-600,pointing-700,video-500,vmci-12000,pvscsi-1000,disk-1000-0,disk-1000-1,ethernet-0,pcipassthrough-0,pcipassthrough-0: timestamp=2023-05-04T12:30:44.404Z[0m
2023-05-04T12:30:37.8145309Z [33;1mVERBOSE: 2023-05-04T12:30:44.404Z [INFO]  provider.terraform-provider-vsphere_v2.2.0_x5: 2023/05/04 12:30:44 VirtualMachine: PCI passthrough device specs created: timestamp=2023-05-04T12:30:44.404Z[0m
2023-05-04T12:30:37.8146197Z [33;1mVERBOSE: 2023-05-04T12:30:44.404Z [INFO]  provider.terraform-provider-vsphere_v2.2.0_x5: 2023/05/04 12:30:44 VirtualMachine: Creating PCI passthrough device specs remove: timestamp=2023-05-04T12:30:44.404Z[0m
2023-05-04T12:30:37.8147014Z [33;1mVERBOSE: 2023-05-04T12:30:44.405Z [INFO]  provider.terraform-provider-vsphere_v2.2.0_x5: 2023/05/04 12:30:44 VirtualMachine: PCI passthrough device specs created: timestamp=2023-05-04T12:30:44.404Z[0m
2023-05-04T12:30:37.8148140Z [33;1mVERBOSE: 2023-05-04T12:30:44.406Z [INFO]  provider.terraform-provider-vsphere_v2.2.0_x5: 2023/05/04 12:30:44 [DEBUG] vsphere_virtual_machine (ID = 421c6936-12e4-218a-2657-53b29194ae39): Final device list: ide-200,ide-201,ps2-300,pci-100,sio-400,keyboard-600,pointing-700,video-500,vmci-12000,pvscsi-1000,disk-1000-0,disk-1000-1,ethernet-0,pcipassthrough-0,pcipassthrough-0: timestamp=2023-05-04T12:30:44.406Z[0m
2023-05-04T12:30:37.8149526Z [33;1mVERBOSE: 2023-05-04T12:30:44.406Z [INFO]  provider.terraform-provider-vsphere_v2.2.0_x5: 2023/05/04 12:30:44 [DEBUG] vsphere_virtual_machine (ID = 421c6936-12e4-218a-2657-53b29194ae39): Final device change spec: (add: *types.VirtualPCIPassthrough at key 0),(add: *types.VirtualPCIPassthrough at key 0): timestamp=2023-05-04T12:30:44.406Z[0m
2023-05-04T12:30:37.8150566Z [33;1mVERBOSE: 2023-05-04T12:30:44.406Z [INFO]  provider.terraform-provider-vsphere_v2.2.0_x5: 2023/05/04 12:30:44 [DEBUG] Reconfiguring virtual machine "/VxRail-Datacenter/vm/VMSK8G02": timestamp=2023-05-04T12:30:44.406Z[0m
2023-05-04T12:30:37.8151312Z [33;1mVERBOSE: 2023-05-04T12:30:44.441Z [ERROR] vertex "module.vm[\"4\"].vsphere_virtual_machine.vm[0]" error: A specified parameter was not correct: deviceChange[1].device.key[0m
````

This looks like the PCI device key is always pointing to pcipassthrough-0 instead of incrementing it.

Is there any ETA when this could be fixed?

--- a/vsphere/internal/virtualdevice/virtual_machine_device_subresource.go
+++ b/vsphere/internal/virtualdevice/virtual_machine_device_subresource.go
@@ -988,6 +988,7 @@ func (c *pciApplyConfig) modifyVirtualPciDevices(devList *schema.Set, op types.V
 					SystemId:                       c.SystemID,
 					VendorId:                       pciDev.VendorId,
 				},
+				Key: c.VirtualDevice.NewKey(),
 			},
 		}
 		vm, err := virtualmachine.FromUUID(c.Client, c.ResourceData.Id())

--- a/vsphere/resource_vsphere_virtual_machine.go
+++ b/vsphere/resource_vsphere_virtual_machine.go
@@ -238,10 +238,11 @@ func resourceVSphereVirtualMachine() *schema.Resource {
 			Elem:        &schema.Resource{Schema: virtualdevice.CdromSubresourceSchema()},
 		},
 		"pci_device_id": {
-			Type:        schema.TypeSet,
-			Optional:    true,
-			Description: "A list of PCI passthrough devices",
-			Elem:        &schema.Schema{Type: schema.TypeString},
+			Type:         schema.TypeSet,
+			Optional:     true,
+			Description:  "A list of PCI passthrough devices",
+			Elem:         &schema.Schema{Type: schema.TypeString},
+			RequiredWith: []string{"host_system_id"},
 		},
 		"clone": {
 			Type:        schema.TypeList,

